### PR TITLE
Feature: Added Support accepting OTLP via Kafka

### PR DESCRIPTION
**Pull request by metapox**
## Which problem is this PR solving?
- Adds two more recognized encodings, otlp-json and otlp-proto.
- https://github.com/jaegertracing/jaeger/issues/3949

## Short description of the changes
-

### CODE REVIEW

File: plugin/storage/kafka/unmarshaller.go [hunk 1]
```diff
@@ -38,10 +40,12 @@ func NewProtobufUnmarshaller() *ProtobufUnmarshaller {
...
 
 // Unmarshal decodes a protobuf byte array to a span
-func (h *ProtobufUnmarshaller) Unmarshal(msg []byte) (*model.Span, error) {
+func (h *ProtobufUnmarshaller) Unmarshal(msg []byte) ([]*model.Span, error) {
 	newSpan := &model.Span{}
+	spans := []*model.Span{newSpan}
 	err := proto.Unmarshal(msg, newSpan)
-	return newSpan, err
+
+	return spans, err
```
Comment (yurishkuro) [inline plugin/storage/kafka/unmarshaller.go:48]:
```
	newSpan := &model.Span{}
	err := proto.Unmarshal(msg, newSpan)
	return []*model.Span{newSpan}, err
```

File: plugin/storage/kafka/options.go [hunk 2]
```diff
@@ -34,6 +34,10 @@ const (
 	EncodingProto = "protobuf"
 	// EncodingZipkinThrift is used for spans encoded as Zipkin Thrift.
```
Comment (yurishkuro) [inline plugin/storage/kafka/options.go:35]:
```suggestion
	// EncodingZipkinThrift is used for spans encoded as Zipkin Thrift array.
```

File: cmd/ingester/app/processor/span_processor.go [hunk 3]
```diff
@@ -62,11 +62,17 @@ func NewSpanProcessor(params SpanProcessorParams) *KafkaSpanProcessor {
...
 	if err != nil {
 		return fmt.Errorf("cannot unmarshall byte array into span: %w", err)
 	}
 
 	// TODO context should be propagated from upstream components
-	return s.writer.WriteSpan(context.TODO(), s.sanitizer(span))
+	for _, span := range spans {
+		err = s.writer.WriteSpan(context.TODO(), s.sanitizer(span))
+		if err != nil {
+			return err
```
Comment (yurishkuro) [inline cmd/ingester/app/processor/span_processor.go:74]: I would log an error here, like `"failed to write %d out of %d spans". 

And this needs to be reflected in the metrics.

File: plugin/storage/kafka/options.go [hunk 4]
```diff
@@ -34,6 +34,10 @@ const (
...
 	EncodingZipkinThrift = "zipkin-thrift"
+	// EncodingOtlpJSON is used for spans encoded as OTLP JSON.
+	EncodingOtlpJSON = "otlp-json"
+	// EncodingOtlpProto is used for spans encoded as OTLP Proto.
+	EncodingOtlpProto = "otlp-proto"
```
Comment (yurishkuro) [inline plugin/storage/kafka/options.go:40]: add to description of encoding flag

File: plugin/storage/kafka/options.go [hunk 5]
```diff
@@ -34,6 +34,10 @@ const (
 	EncodingProto = "protobuf"
 	// EncodingZipkinThrift is used for spans encoded as Zipkin Thrift.
 	EncodingZipkinThrift = "zipkin-thrift"
+	// EncodingOtlpJSON is used for spans encoded as OTLP JSON.
```
Comment (yurishkuro) [inline plugin/storage/kafka/options.go:37]: clarify which Protobuf type is expected in the Kafka message

**Review by yurishkuro — COMMENTED** [review]
please add tests for all code changes

**Comment by metapox** [comment]
thank you for review.
Can you please review it again?

**Comment by yurishkuro** [comment]
please resolve merge conflicts

**Comment by metapox** [comment]
Sorry,,,,
By mistake, I synchronized this branch with the main branch....

A new PR has been created because a PR with no commits cannot be reopened.
https://github.com/jaegertracing/jaeger/pull/4333

**Pull request closed without merging**